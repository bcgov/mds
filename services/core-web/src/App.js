import React, { Fragment } from "react";
import { BrowserRouter } from "react-router-dom";
import { hot } from "react-hot-loader";
import Routes from "./routes/Routes";
import ModalWrapper from "@/components/common/wrappers/ModalWrapper";
import ScrollToTopWrapper from "@/components/common/wrappers/ScrollToTopWrapper";
import configureStore from "./store/configureStore";
import { MatomoLinkTracing } from "../common/utils/trackers";

export const store = configureStore();

const App = () => (
  <BrowserRouter basename={process.env.BASE_PATH}>
    <ScrollToTopWrapper>
      <Fragment>
        <MatomoLinkTracing />
        <Routes />
        <ModalWrapper />
      </Fragment>
    </ScrollToTopWrapper>
  </BrowserRouter>
);

export default hot(module)(App);
