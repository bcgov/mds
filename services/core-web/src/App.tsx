import React, { useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import { hot } from "react-hot-loader";
import Routes from "./routes/Routes";
import ModalWrapper from "@/components/common/wrappers/ModalWrapper";
import DocumentViewer from "@/components/syncfusion/DocumentViewer";
import ScrollToTopWrapper from "@/components/common/wrappers/ScrollToTopWrapper";
import configureStore from "./store/configureStore";

export const store = configureStore();

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useKey = (test: (e: KeyboardEvent) => boolean, ref) => {
  useEffect(() => {
    const hotKeyPress = (event: KeyboardEvent) => {
      if (test(event)) {
        event.preventDefault();
        ref.current.focus();
        return;
      }
    };
    document.addEventListener("keydown", hotKeyPress);
    return () => document.removeEventListener("keydown", hotKeyPress);
  });
};

const App = () => {
  return (
    <BrowserRouter basename={process.env.BASE_PATH}>
      <ScrollToTopWrapper>
        <>
          <Routes />
          <ModalWrapper />
          <DocumentViewer />
        </>
      </ScrollToTopWrapper>
    </BrowserRouter>
  );
};

export default hot(module)(App);
