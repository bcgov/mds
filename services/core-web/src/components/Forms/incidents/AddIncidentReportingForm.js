import React from "react";
import PropTypes from "prop-types";
import { Field, reduxForm } from "redux-form";
import { Form, Col, Row } from "antd";
import {
  required,
  requiredList,
  email,
  number,
  phoneNumber,
  maxLength,
  dateNotInFuture,
} from "@common/utils/Validate";
import { normalizePhone } from "@common/utils/helpers";
import * as FORM from "@/constants/forms";
import { renderConfig } from "@/components/common/config";
import CustomPropTypes from "@/customPropTypes";

const propTypes = {
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
  inspectors: CustomPropTypes.groupOptions.isRequired,
  incidentCategoryCodeOptions: CustomPropTypes.options.isRequired,
};

export const AddIncidentReportingForm = (props) => (
  <div>
    <Form layout="vertical">
      <Row gutter={48}>
        <Col>
          {props.initialValues.mine_incident_id_year && (
            <h4>{`Ministry Incident No. ${props.initialValues.mine_incident_report_no}`}</h4>
          )}
          {props.initialValues.mms_inspector_initials ? (
            <span style={{ float: "right" }}>
              {`MMS Inspector Initials ${props.initialValues.mms_inspector_initials}`}
            </span>
          ) : (
            ""
          )}
          <Form.Item>
            <Field
              id="categories"
              name="categories"
              label="Incident type(s)*"
              placeholder="Select incident type(s)"
              component={renderConfig.MULTI_SELECT}
              validate={[requiredList]}
              data={props.incidentCategoryCodeOptions}
            />
          </Form.Item>
          <Form.Item>
            <Field
              id="reported_to_inspector_party_guid"
              name="reported_to_inspector_party_guid"
              label="Incident reported to*"
              placeholder="Search for inspector"
              component={renderConfig.GROUPED_SELECT}
              format={null}
              validate={[required]}
              data={props.inspectors}
            />
          </Form.Item>
          <Form.Item>
            <Field
              id="responsible_inspector_party_guid"
              name="responsible_inspector_party_guid"
              label="Inspector responsible*"
              component={renderConfig.GROUPED_SELECT}
              format={null}
              placeholder="Search for responsible inspector"
              validate={[required]}
              data={props.inspectors}
            />
          </Form.Item>
          <h4>Reporter Details</h4>
          <Form.Item>
            <Field
              id="reported_by_name"
              name="reported_by_name"
              label="Reported by*"
              placeholder="Enter name of reporter"
              component={renderConfig.FIELD}
              validate={[required]}
            />
          </Form.Item>
          <Row gutter={16}>
            <Col md={12} xs={24}>
              <Form.Item>
                <Field
                  id="reported_by_phone_no"
                  name="reported_by_phone_no"
                  label="Phone number"
                  placeholder="xxx-xxx-xxxx"
                  component={renderConfig.FIELD}
                  validate={[phoneNumber, maxLength(12)]}
                  normalize={normalizePhone}
                />
              </Form.Item>
            </Col>
            <Col md={12} xs={24}>
              <Form.Item>
                <Field
                  id="reported_by_phone_ext"
                  name="reported_by_phone_ext"
                  label="Phone extension"
                  placeholder="xxxxxx"
                  component={renderConfig.FIELD}
                  validate={[number, maxLength(6)]}
                />
              </Form.Item>
            </Col>
          </Row>
          <Form.Item>
            <Field
              id="reported_by_email"
              name="reported_by_email"
              label="Email"
              placeholder="example@domain.com"
              component={renderConfig.FIELD}
              validate={[email]}
            />
          </Form.Item>
          <Row gutter={16}>
            <Col md={12} xs={24}>
              <Form.Item>
                <Field
                  id="reported_date"
                  name="reported_date"
                  label="Reported Date*"
                  placeholder="Please select date"
                  component={renderConfig.DATE}
                  validate={[required, dateNotInFuture]}
                />
              </Form.Item>
            </Col>
            <Col md={12} xs={24}>
              <Form.Item>
                <Field
                  id="reported_time"
                  name="reported_time"
                  label="Reported Time*"
                  placeholder="Please select time"
                  component={renderConfig.TIME}
                  validate={[required]}
                  fullWidth
                />
              </Form.Item>
            </Col>
          </Row>
        </Col>
      </Row>
    </Form>
  </div>
);

AddIncidentReportingForm.propTypes = propTypes;

export default reduxForm({
  form: FORM.MINE_INCIDENT,
  destroyOnUnmount: false,
  forceUnregisterOnUnmount: true,
  touchOnBlur: true,
})(AddIncidentReportingForm);
