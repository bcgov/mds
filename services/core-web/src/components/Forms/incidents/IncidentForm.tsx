import React, { FC, useState } from "react";
import { useParams, withRouter } from "react-router-dom";
import { compose } from "redux";
import { useDispatch, useSelector } from "react-redux";
import { change, formValueSelector, getFormValues, InjectedFormProps, reduxForm } from "redux-form";
import { Form } from "@ant-design/compatible";
import "@ant-design/compatible/assets/index.css";
import { Button, Col, Row } from "antd";
import { IMineIncident } from "@mds/common";
import { getDropdownInspectors } from "@mds/common/redux/selectors/partiesSelectors";
import {
  getDangerousOccurrenceSubparagraphOptions,
  getDropdownIncidentDeterminationOptions,
  getDropdownIncidentFollowupActionOptions,
  getDropdownIncidentStatusCodeOptions,
  getIncidentStatusCodeHash,
} from "@mds/common/redux/selectors/staticContentSelectors";
import * as FORM from "@/constants/forms";
import {
  documentNameColumn,
  uploadDateColumn,
  uploadedByColumn,
} from "@/components/common/DocumentColumns";
import IncidentFormDocuments from "@/components/Forms/incidents/incidentFormDocuments";
import IncidentFormInternalDocumentComments from "@/components/Forms/incidents/IncidentFormInternalDocumentComments";
import IncidentFormInitialReport from "@/components/Forms/incidents/incidentFormInitialReport";
import IncidentFormUpdateIncidentStatus from "@/components/Forms/incidents/IncidentFormUpdateIncidentStatus";
import IncidentFormMinistryFollowup from "@/components/Forms/incidents/IncidentFormMinistryFollowup";
import { getMineIncident } from "@mds/common/redux/reducers/incidentReducer";
import { removeDocumentFromMineIncident } from "@mds/common/redux/actionCreators/incidentActionCreator";

export const INITIAL_INCIDENT_DOCUMENTS_FORM_FIELD = "initial_incident_documents";
export const FINAL_REPORT_DOCUMENTS_FORM_FIELD = "final_report_documents";
export const INTERNAL_MINISTRY_DOCUMENTS_FORM_FIELD = "internal_ministry_documents";

export const documentColumns = [documentNameColumn(), uploadDateColumn(), uploadedByColumn()];

export const formatDocumentRecords = (documents) =>
  documents?.map((doc) => ({ ...doc, key: doc.mine_document_guid }));

interface IncidentFormProps {
  initialValues: IMineIncident;
  isEditMode: boolean;
  isNewIncident: boolean;
  incident: IMineIncident;
  handlers: {
    handleSaveData: () => void;
    handleFetchData: () => void;
  };
}

export const IncidentForm: FC<IncidentFormProps & InjectedFormProps> = (props) => {
  const { mineIncidentGuid, mineGuid } = useParams<{
    mineGuid: string;
    mineIncidentGuid: string;
  }>();

  const { isEditMode, handlers: parentHandlers } = props;
  const isNewIncident = Boolean(!mineIncidentGuid);

  const dispatch = useDispatch();

  const incidentDeterminationOptions = useSelector(getDropdownIncidentDeterminationOptions);
  const incidentStatusCodeHash = useSelector(getIncidentStatusCodeHash);
  const dangerousOccurenceSubparagraphOptions = useSelector(
    getDangerousOccurrenceSubparagraphOptions
  );
  const incidentFollowUpActionOptions = useSelector(getDropdownIncidentFollowupActionOptions);
  const inspectorOptions = useSelector(getDropdownInspectors) || [];
  const selector = formValueSelector(FORM.ADD_EDIT_INCIDENT);
  const documents = useSelector((state) => selector(state, "documents")) || [];
  const formValues = useSelector((state) => getFormValues(FORM.ADD_EDIT_INCIDENT)(state)) || {};
  const dropdownIncidentStatusCodeOptions = useSelector(getDropdownIncidentStatusCodeOptions);
  const isPristine = useSelector((state) => state.form[FORM.ADD_EDIT_INCIDENT]?.pristine);

  const incident = useSelector(getMineIncident);

  const [uploadedFiles, setUploadedFiles] = useState([]);

  const onFileLoad = (fileName, document_manager_guid, documentTypeCode, documentFormField) => {
    const updatedUploadedFiles = [
      ...uploadedFiles,
      {
        document_name: fileName,
        document_manager_guid,
        mine_incident_document_type_code: documentTypeCode,
      },
    ];
    setUploadedFiles(updatedUploadedFiles);
    return dispatch(change(FORM.ADD_EDIT_INCIDENT, documentFormField, updatedUploadedFiles));
  };

  const onRemoveFile = (err, fileItem, documentFormField) => {
    const updatedUploadedFiles = uploadedFiles.filter(
      (doc) => doc.document_manager_guid !== fileItem.serverId
    );
    setUploadedFiles(updatedUploadedFiles);
    return dispatch(change(FORM.ADD_EDIT_INCIDENT, documentFormField, updatedUploadedFiles));
  };

  const handleDeleteDocument = (event, mineDocumentGuid: string) => {
    if (mineGuid && props.incident.mine_incident_guid && mineDocumentGuid) {
      return dispatch(
        removeDocumentFromMineIncident(
          mineGuid,
          props.incident.mine_incident_guid,
          mineDocumentGuid
        )
      ).then(() => props.handlers.handleFetchData());
    }
    return null;
  };

  const renderEditSaveControls = () => {
    return (
      <div className="right center-mobile violet">
        {isEditMode && (
          <Button
            id="mine-incident-submit"
            className="full-mobile right"
            type="primary"
            htmlType="submit"
            loading={props.submitting}
            disabled={props.submitting}
          >
            {isNewIncident ? "Create Incident" : "Save Changes"}
          </Button>
        )}
      </div>
    );
  };

  return (
    <Form layout="vertical" onSubmit={props.handleSubmit(parentHandlers.handleSaveData)}>
      <Col span={24}>
        Is Edit - {isEditMode.toString()}
        <IncidentFormUpdateIncidentStatus
          incident={props.incident}
          dropdownIncidentStatusCodeOptions={dropdownIncidentStatusCodeOptions}
          incidentStatusCodeHash={incidentStatusCodeHash}
          isEditMode={isEditMode}
          formValues={formValues}
          pristine={isPristine}
        />
      </Col>
      <Row>
        <Col span={24}>{renderEditSaveControls()}</Col>
        <Col span={16} offset={4}>
          <IncidentFormInitialReport incident={props.incident} isEditMode={isEditMode} />
          <br />
          <IncidentFormDocuments
            incidentGuid={incident.mine_incident_guid}
            documents={documents}
            isEditMode={isEditMode}
            onFileLoad={onFileLoad}
            onDeleteDocument={handleDeleteDocument}
            onRemoveFile={onRemoveFile}
          />
          <br />
          <IncidentFormMinistryFollowup
            isEditMode={isEditMode}
            incidentDeterminationOptions={incidentDeterminationOptions}
            incidentFollowUpActionOptions={incidentFollowUpActionOptions}
            dangerousOccurenceSubparagraphOptions={dangerousOccurenceSubparagraphOptions}
            inspectorOptions={inspectorOptions}
          />
          <br />
          <IncidentFormInternalDocumentComments
            documents={documents}
            incident={props.incident}
            isEditMode={isEditMode}
            onFileLoad={onFileLoad}
            onDeleteDocument={handleDeleteDocument}
            formValues={formValues}
            onRemoveFile={onRemoveFile}
          />
        </Col>
        <Col span={24}>
          <br />
          {renderEditSaveControls()}
        </Col>
      </Row>
    </Form>
  );
};

export default compose(
  withRouter,
  reduxForm({
    form: FORM.ADD_EDIT_INCIDENT,
    enableReinitialize: true,
    touchOnBlur: true,
    touchOnChange: true,
    destroyOnUnmount: true,
  })
)(IncidentForm);
