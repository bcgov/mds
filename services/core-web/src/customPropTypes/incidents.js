import { PropTypes, shape } from "prop-types";

export const incident = shape({
  mine_incident_guid: PropTypes.string,
  mine_incident_report_no: PropTypes.string,
  mine_incident_id_year: PropTypes.number,
  mine_guid: PropTypes.string,
  incident_timestamp: PropTypes.string,
  incident_description: PropTypes.string,
  reported_timestamp: PropTypes.string,
  reported_by: PropTypes.string,
  reported_by_role: PropTypes.string,
  followup_type_code: PropTypes.string,
  followup_inspection_no: PropTypes.string,
  closing_report_summary: PropTypes.string,
  mms_inspector_initials: PropTypes.string,
});

export const incidentFollowupType = shape({
  mine_incident_followup_investigation_type_code: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  display_order: PropTypes.number.isRequired,
});

export const incidentSearchInitialValues = shape({
  region: PropTypes.arrayOf(PropTypes.string),
  codes: PropTypes.arrayOf(PropTypes.string),
  major: PropTypes.string,
  year: PropTypes.string,
  incident_status: PropTypes.arrayOf(PropTypes.string),
  determination: PropTypes.arrayOf(PropTypes.string),
  search: PropTypes.string,
});

export const incidentPageData = shape({
  records: PropTypes.arrayOf(incident),
  current_page: PropTypes.number,
  items_per_page: PropTypes.number,
  total: PropTypes.number,
  total_pages: PropTypes.number,
});

export default incident;
