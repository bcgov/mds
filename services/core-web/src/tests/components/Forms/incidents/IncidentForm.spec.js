import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import { IncidentForm } from "@/components/Forms/incidents/IncidentForm";
import * as MOCK from "@/tests/mocks/dataMocks";
import { store } from "@/App";

const dispatchProps = {};
const props = {};

const setupDispatchProps = () => {
  dispatchProps.change = jest.fn(() => Promise.resolve());
};

const setupProps = () => {
  props.incident = MOCK.INCIDENT;
  props.formValues = {
    initial_incident_documents: [],
    final_report_documents: [],
    internal_ministry_documents: [],
  };
  props.handlers = {
    deleteDocument: jest.fn(() => Promise.resolve()),
  };
  props.incidentCategoryCodeOptions = [];
  props.handleSubmit = jest.fn();
  props.isEditMode = false;
  props.match = {
    mineGuid: "448014a5-981f-47b8-8687-4963666776b8",
    mineIncidentGuid: "668014a5-981f-47b8-8687-4963666776b9",
  };
  props.documents = [
    {
      mine_incident_document_type_code: "INI",
    },
  ];
  props.incidentFollowUpActionOptions = [
    {
      mine_incident_followup_investigation_type: "TEST",
    },
  ];
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("IncidentForm", () => {
  it("renders properly", () => {
    const component = shallow(
      <Provider store={store}>
        <IncidentForm {...dispatchProps} {...props} />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
