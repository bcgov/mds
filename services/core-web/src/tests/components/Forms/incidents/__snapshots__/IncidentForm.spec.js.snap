// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncidentForm renders properly 1`] = `
<Form
  colon={true}
  hideRequiredMark={false}
  layout="vertical"
  onSubmit={[Function]}
>
  <Col
    span={24}
  />
  <Row>
    <Col
      span={24}
    >
      <div
        className="right center-mobile violet"
      >
        <Button
          block={false}
          className="full-mobile violet violet-border"
          ghost={false}
          htmlType="button"
          id="mine-incident-edit"
          loading={false}
          type="secondary"
        >
          <img
            alt="pencil"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Â Edit Incident
        </Button>
      </div>
    </Col>
    <Col
      offset={4}
      span={16}
    >
      <Row>
        <Col
          span={24}
        >
          <Title
            id="initial-report"
            level={3}
          >
            Initial Report
          </Title>
          <h4>
            Reporter Details
          </h4>
          <Paragraph>
            Enter all available details about the reporter of this incident.
          </Paragraph>
          <Row
            gutter={
              Array [
                16,
              ]
            }
          >
            <Col
              md={10}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="* Reported by"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="reported_by_name"
                  name="reported_by_name"
                  placeholder="Enter name of reporter..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={10}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="* Phone number"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="reported_by_phone_no"
                  name="reported_by_phone_no"
                  normalize={[Function]}
                  placeholder="xxx-xxx-xxxx"
                  validate={
                    Array [
                      [Function],
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={4}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Ext."
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="reported_by_phone_ext"
                  name="reported_by_phone_ext"
                  placeholder="xxxxxx"
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={10}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="* Email"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="reported_by_email"
                  name="reported_by_email"
                  placeholder="example@domain.com"
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
          </Row>
          <br />
        </Col>
        <Col
          span={24}
        >
          <h4
            id="incident-details"
          >
            Incident Details
          </h4>
          <Paragraph>
            Enter more information regarding the reported incident. Some fields may be marked as optional but help the ministry understand the nature of the incident, please consider including them.
          </Paragraph>
          <Row
            gutter={
              Array [
                16,
              ]
            }
          >
            <Col
              span={24}
            >
              <FormItem
                hasFeedback={false}
                label="* Incident type(s)"
              >
                <Field
                  component={[Function]}
                  data={Array []}
                  disabled={true}
                  id="categories"
                  name="categories"
                  placeholder="Select incident type(s)..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="* Incident date"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="incident_date"
                  name="incident_date"
                  placeholder="Please select date..."
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="* Incident time"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  fullWidth={true}
                  id="incident_time"
                  name="incident_time"
                  placeholder="Please select time..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Proponent incident number (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="proponent_incident_no"
                  name="proponent_incident_no"
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Number of injuries (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="number_of_injuries"
                  name="number_of_injuries"
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Number of fatalities (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="number_of_fatalities"
                  name="number_of_fatalities"
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Were emergency services called? (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="emergency_services_called"
                  name="emergency_services_called"
                  placeholder="Please choose one..."
                />
              </FormItem>
            </Col>
            <Col
              span={24}
            >
              <FormItem
                hasFeedback={false}
                label="* Description of incident"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="incident_description"
                  name="incident_description"
                  placeholder="Provide a detailed description of the incident..."
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              span={24}
            >
              <FormItem
                hasFeedback={false}
                label="Immediate measures taken (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="immediate_measures_taken"
                  name="immediate_measures_taken"
                  placeholder="Provide a detailed description of any immediate measures taken..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              span={24}
            >
              <FormItem
                hasFeedback={false}
                label="If any injuries, please describe (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="injuries_description"
                  name="injuries_description"
                  placeholder="Provide a detailed description of any injuries..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Divider />
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="JOHSC/Worker Rep Name (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="johsc_worker_rep_name"
                  name="johsc_worker_rep_name"
                  placeholder="Enter name of representative..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Was this person contacted? (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="johsc_worker_rep_contacted"
                  name="johsc_worker_rep_contacted"
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="JOHSC/Management Rep Name (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="johsc_management_rep_name"
                  name="johsc_management_rep_name"
                  placeholder="Enter name of representative..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Was this person contacted? (optional)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="johsc_management_rep_contacted"
                  name="johsc_management_rep_contacted"
                />
              </FormItem>
              <br />
            </Col>
          </Row>
          <br />
        </Col>
      </Row>
      <br />
      <Row>
        <Col
          span={24}
        >
          <Title
            id="documentation"
            level={3}
          >
            Documentation
          </Title>
          <Row>
            <Col
              md={12}
              xs={24}
            >
              <h4>
                Upload Initial Notification Documents
              </h4>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <div
                className="right center-mobile"
              >
                <Button
                  block={false}
                  className="full-mobile violet violet-border"
                  ghost={false}
                  htmlType="button"
                  id="mine-incident-add-documentation"
                  loading={false}
                  type="secondary"
                >
                  + Add Documentation
                </Button>
              </div>
            </Col>
          </Row>
          <br />
          <h4>
            Incident Documents
          </h4>
          <br />
          <Paragraph>
            Please upload any initial notifications that will provide context with this incident report.
          </Paragraph>
        </Col>
        <Col
          span={24}
        >
          <DocumentTable
            additionalColumnProps={Array []}
            documentColumns={
              Array [
                Object {
                  "dataIndex": "document_name",
                  "render": [Function],
                  "sorter": [Function],
                  "title": "Name",
                },
                Object {
                  "dataIndex": "update_user",
                  "render": [Function],
                  "title": "Uploaded",
                },
                Object {
                  "dataIndex": "upload_date",
                  "render": [Function],
                  "sorter": [Function],
                  "title": "Upload Date",
                },
              ]
            }
            documentParent="Mine Incident"
            documents={
              Array [
                Object {
                  "key": undefined,
                  "mine_incident_document_type_code": "INI",
                },
              ]
            }
            excludedColumnKeys={Array []}
            isViewOnly={false}
            removeDocument={false}
          />
          <br />
        </Col>
        <Col
          span={24}
        >
          <h4
            id="final-report"
          >
            Final Report
          </h4>
          <br />
        </Col>
        <Col
          span={24}
        >
          <DocumentTable
            additionalColumnProps={Array []}
            documentColumns={
              Array [
                Object {
                  "dataIndex": "document_name",
                  "render": [Function],
                  "sorter": [Function],
                  "title": "Name",
                },
                Object {
                  "dataIndex": "update_user",
                  "render": [Function],
                  "title": "Uploaded",
                },
                Object {
                  "dataIndex": "upload_date",
                  "render": [Function],
                  "sorter": [Function],
                  "title": "Upload Date",
                },
              ]
            }
            documentParent="Mine Incident"
            documents={Array []}
            excludedColumnKeys={Array []}
            isViewOnly={false}
            removeDocument={false}
          />
          <br />
        </Col>
        <Col
          span={24}
        >
          <Empty
            description={
              <div
                className="center"
              >
                <Paragraph
                  strong={true}
                >
                  This incident requires a final investigation report.
                </Paragraph>
                <Paragraph>
                  You determined that this incident was a dangerous occurence. Please add your final report documentation by clicking below.
                </Paragraph>
                <ForwardRef(InternalButton)
                  block={false}
                  ghost={false}
                  htmlType="button"
                  loading={false}
                  type="primary"
                >
                  Add Final Report
                </ForwardRef(InternalButton)>
              </div>
            }
            image={<Simple />}
          />
        </Col>
      </Row>
      <br />
      <Row>
        <Col
          span={24}
        >
          <Title
            id="ministry-follow-up"
            level={3}
          >
            Ministry Follow Up
          </Title>
          <h4>
            Dangerous Occurence Determination
          </h4>
          <Row
            gutter={16}
          >
            <Col
              span={24}
            >
              <FormItem
                hasFeedback={false}
                label="Inspector's Determination"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="determination_type_code"
                  name="determination_type_code"
                  placeholder="Select determination..."
                  validate={
                    Array [
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              span={24}
            >
              <h4>
                Verbal Notification
              </h4>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Verbal notification must be provided within 4 hours of the reportable  incident. Was verbal notification of the incident provided through the Mine Incident Reporting Line (1-888-348-0299)? (Yes/No)"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="verbal_notification_provided"
                  name="verbal_notification_provided"
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Date and time"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="verbal_notification_timestamp"
                  name="verbal_notification_timestamp"
                  placeholder="Please select date"
                  showTime={true}
                  validate={
                    Array [
                      [Function],
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              span={24}
            >
              <h4>
                Follow-Up Information
              </h4>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Incident reported to"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  format={null}
                  id="reported_to_inspector_party_guid"
                  name="reported_to_inspector_party_guid"
                  placeholder="Search for inspector..."
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Was this person contacted?"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="reported_to_inspector_contacted"
                  name="reported_to_inspector_contacted"
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Inspector responsible"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  format={null}
                  id="responsible_inspector_party_guid"
                  name="responsible_inspector_party_guid"
                  placeholder="Search for responsible inspector..."
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Was there a follow-up inspection?"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="followup_inspection"
                  name="followup_inspection"
                />
              </FormItem>
            </Col>
            <Col
              md={12}
              xs={24}
            >
              <FormItem
                hasFeedback={false}
                label="Follow-up inspection date"
              >
                <Field
                  component={[Function]}
                  disabled={true}
                  id="followup_inspection_date"
                  name="followup_inspection_date"
                  placeholder="Please select date..."
                  validate={
                    Array [
                      [Function],
                      [Function],
                    ]
                  }
                />
              </FormItem>
            </Col>
            <Col
              span={24}
            >
              <FormItem
                hasFeedback={false}
                label="Was it escalated to EMLI investigation?"
              >
                <Field
                  component={[Function]}
                  customOptions={
                    Array [
                      Object {
                        "mine_incident_followup_investigation_type": "TEST",
                      },
                    ]
                  }
                  disabled={true}
                  id="followup_investigation_type_code"
                  name="followup_investigation_type_code"
                  validate={[Function]}
                />
              </FormItem>
            </Col>
            <Col
              span={24}
            >
              <FormItem
                hasFeedback={false}
                label="Mine manager's recommendations"
              >
                <FieldArray
                  component={[Function]}
                  handlers={
                    Object {
                      "deleteDocument": [MockFunction],
                    }
                  }
                  id="recommendations"
                  isEditMode={false}
                  name="recommendations"
                />
              </FormItem>
            </Col>
          </Row>
        </Col>
      </Row>
      <br />
      <Row>
        <Col
          span={24}
        >
          <Title
            id="internal-documents"
            level={3}
          >
            <ForwardRef(LockOutlined)
              className="violet"
            />
             
            Internal Documents and Comments (Ministry Visible Only)
          </Title>
          <Divider />
          <div
            className="center"
          >
            <Empty
              description={
                <Paragraph
                  className="center padding-md--top"
                  strong={true}
                >
                  The internal ministry documentation section will be displayed after this incident is created.
                </Paragraph>
              }
              image={<Simple />}
            />
          </div>
        </Col>
      </Row>
    </Col>
    <Col
      span={24}
    >
      <br />
      <div
        className="right center-mobile violet"
      >
        <Button
          block={false}
          className="full-mobile violet violet-border"
          ghost={false}
          htmlType="button"
          id="mine-incident-edit"
          loading={false}
          type="secondary"
        >
          <img
            alt="pencil"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          Â Edit Incident
        </Button>
      </div>
    </Col>
  </Row>
</Form>
`;
